### PR TITLE
Notify user if some enabled repository is unknown to Leapp

### PR DESCRIPTION
**Pull request by artmello**

### CODE REVIEW

File: repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py [hunk 1]
```diff
@@ -19,18 +20,46 @@ class RpmScanner(Actor):
     def process(self):
         output = get_installed_rpms()
 
+        pkg_repo = {}
+        header = True
+        tmp = None
+        for entry in run(['yum', 'list', 'installed'], split=True)['stdout']:
+            # Skip command output header...
+            if header:
+                if entry == 'Installed Packages':
```
Comment (pirat89) [inline repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py]: Rather try to find different way as I would afraid of translations in other languages on other machines.

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 2]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+                RepositoriesMap,
+                RepositoriesFacts,
+                RepositoriesBlacklisted,
+                RepositoriesUsage)
+    produces = (TargetRepositories, Report,)
     tags = (IPUWorkflowTag, ChecksPhaseTag)
 
+
+    def report_skipped_repos(self, repos, pkgs):
+        title = 'Some enabled RPM repositories are not compatible with Leapp'
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]:
```suggestion
        title = 'Some enabled RPM repositories are unknown to Leapp'
```

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 3]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+        summary_data = []
+        summary_data.append('Following repositories are unknown to Leapp:')
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]:
```suggestion
        summary_data.append('The following repositories with Red Hat-signed packages are unknown to Leapp:')
```

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 4]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+        summary_data.extend(['- ' + r for r in repos])
+        summary_data.append('And the following installed packages from those repositories may not be upgraded:')
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]:
```suggestion
        summary_data.append('And the following packages installed from those repositories may not be upgraded:')
```

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 5]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+        summary_data.extend(['- ' + p for p in pkgs])
+        summary = '\n'.join(summary_data)
+        reporting.report_generic(title=title, summary=summary, severity='low')
+        if True: #is_verbose():
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]: Change or remove this line.

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 6]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+            api.show_message('\n'.join([title, summary]))
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]: I don't like using this function much, as the only thing it does is to print an message to stdout. It does not go to audit or logs.
Related: https://github.com/oamg/leapp/pull/497

File: repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py [hunk 7]
```diff
@@ -16,10 +21,29 @@ class SetupTargetRepos(Actor):
...
+
+    def report_skipped_repos(self, repos, pkgs):
+        title = 'Some enabled RPM repositories are not compatible with Leapp'
+        summary_data = []
+        summary_data.append('Following repositories are unknown to Leapp:')
+        summary_data.extend(['- ' + r for r in repos])
+        summary_data.append('And the following installed packages from those repositories may not be upgraded:')
+        summary_data.extend(['- ' + p for p in pkgs])
+        summary = '\n'.join(summary_data)
+        reporting.report_generic(title=title, summary=summary, severity='low')
```
Comment (bocekm) [inline repos/system_upgrade/el7toel8/actors/setuptargetrepos/actor.py]: Add a remediation: "You can file a request to add this repository to the scope of in-place upgrades by filing a support ticket"

File: repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py (hunk 1 above)
Comment (artmello) [reply repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py]: I have played a little bit with yum module. This example using returnPackages returns "installed" as repoid for all packages. The output of "yum list installed" returns "installed" for all packages installed together with system, but shows a valid repoid for packages not installed by anaconda.

**Comment by artmello** [comment]
Hey @bocekm, I hope all your comments are addressed with last commit. Let me know how does this looks like to you.

**Comment by bocekm** [comment]
I'm on it. Sorry for the delay.

File: repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py (hunk 1 above)
Comment (bocekm) [reply repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py]: It seems there's a bug in the yum library. Even this does not work:
```python
import yum
yum_base = yum.YumBase()
for pkg in yum_base.doPackageLists().installed:
  print(pkg['repoid'])
```
It prints just 'installed'.

Comment (bocekm) [reply repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py]: Aha, found it, we shouldn't be looking at `repoid` but `ui_from_repo`, based on https://github.com/rpm-software-management/yum/blob/master/output.py#L734.
So:
```python
import yum
yum_base = yum.YumBase()
for pkg in yum_base.doPackageLists().installed:
  print(pkg['ui_from_repo'])
```
... gives us the right stuff (e.g. *@anaconda/7.6*, *@rhel-7-server-extras-rpms*, etc.)

File: repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py [hunk 8]
```diff
@@ -16,21 +17,56 @@ class RpmScanner(Actor):
     produces = (InstalledRPM,)
     tags = (IPUWorkflowTag, FactsPhaseTag)
 
+    def get_package_repository_data(self):
```
[the remark this answers is not in the document]
Comment (artmello) [reply repos/system_upgrade/el7toel8/actors/rpmscanner/actor.py]: Code changed, let me know how does it looks like to you know.

**Pull request merged**